system/jlib/jtrace.cpp

```diff
@@ -1350,6 +1359,59 @@ std::unique_ptr<opentelemetry::sdk::trace::SpanProcessor> CTraceManager::createP
     return opentelemetry::sdk::trace::SimpleSpanProcessorFactory::Create(std::move(exporter));
 }
 
+static std::unique_ptr<opentelemetry::sdk::trace::Sampler> createSampler(IPropertyTree * samplerTree)
+{
+    std::unique_ptr<opentelemetry::sdk::trace::Sampler> sampler;
+
+    if (samplerTree)
+    {
+        const char * samplerType = samplerTree->queryProp("@type");
+        if (!isEmptyString(samplerType))
+        {
+            const char * samplerArgument = samplerTree->queryProp("@argument");
+            if (strcmp("AlwaysOff", samplerType)==0)
+            {
+                sampler.reset(new opentelemetry::sdk::trace::AlwaysOffSampler());
+            }
+            else if (strcmp("AlwaysOn", samplerType)==0)
+            {
+                sampler.reset(new opentelemetry::sdk::trace::AlwaysOnSampler());
+            }
+            else if (strcmp("Ratio", samplerType)==0)
+            {
+                size_t pos;
+                double ratio = std::stod(samplerArgument, &pos);
```

Comment:
> const char * samplerArgument = samplerTree->queryProp("@argument");

could move to here, as only use of argument, and do:
```
    double ratio = samplerTree->getPropReal("@argument");
```

Reply:
Agreed, it's meant to be a generic argument for future support, but in reality the ratio is likely to be the only  use, will use your suggestion